lib/PuppeteerSharp.Tests/Puppeteer/PuppeteerLaunchTests.cs

```diff
@@ -38,5 +42,53 @@ public PuppeteerLaunchTests()
 
             await browser.CloseAsync();
         }
+
+        [Fact]
+        public async Task NetworkRedirectsShouldReportSecurityDetails()
+        {
+            var options = TestConstants.DefaultBrowserOptions();
+            options.IgnoreHTTPSErrors = true;
+
+            var browser = await PuppeteerSharp.Puppeteer.LaunchAsync(options, TestConstants.ChromiumRevision);
+            var page = await browser.NewPageAsync();
+            
+            var responses = new List<Response>();
+            page.ResponseCreated += (sender, e) => responses.Add(e.Response);
+
+            await page.GoToAsync(TestConstants.ServerUrl + "/plzredirect");
+            Assert.Equal(2, responses.Count);
+            Assert.Equal(HttpStatusCode.Redirect, responses[0].Status);
+            var securityDetails = responses[0].SecurityDetails;
+            Assert.Equal("TLS 1.2", securityDetails.Protocol);
+
+            await page.CloseAsync();
+            await browser.CloseAsync();
+        }
+
+        [Fact(Skip = "https://github.com/kblok/puppeteer-sharp/issues/76")]
+        public async Task ShouldRejectAllPromisesWhenBrowserIsClosed()
+        {
+            var browser = await PuppeteerSharp.Puppeteer.LaunchAsync(TestConstants.DefaultBrowserOptions(), TestConstants.ChromiumRevision);
+            var page = await browser.NewPageAsync();
+
+            var neverResolves = page.EvaluateHandle("() => new Promise(r => {})");
+            await browser.CloseAsync();
+
+            await neverResolves;
+            var exception = await Assert.ThrowsAsync<Exception>(() => neverResolves);
+            Assert.Contains("Protocol error", exception.Message);
+        }
+
+        [Fact]
+        public async Task ShouldRejectIfExecutablePathIsInvalid()
+        {
+            var options = TestConstants.DefaultBrowserOptions();
+            options.ExecutablePath = "random-invalid-path";
+
+            var exception = await Assert.ThrowsAsync<Exception>(() => PuppeteerSharp.Puppeteer.LaunchAsync(options, TestConstants.ChromiumRevision));
+            Assert.Equal("Failed to create connection", exception.Message);
```

Comment:
the exception is thrown from https://github.com/kblok/puppeteer-sharp/blob/master/lib/PuppeteerSharp/Launcher.cs#L122, this is because the `WaitForEndpoint` method calls the `Start` method on the process object